test/versioning/npm.spec.js

```diff
@@ -123,4 +123,88 @@ describe('semver.getNewValue()', () => {
+      ['=1', '1.0.0', '1.2.3', '=1 <=1.2.3'], // '>=1.0.0 <2.0.0'
+
+      ['1.0.0 || 1.1.0', '1.0.0', '1.2.3', '1.0.0 || 1.2.3'],
+      ['1.0.0', '1.0.0', '1.2.3', '1.2.3'],
+      ['1.2', '1.0.0', '1.2.3', '1.2 <=1.2.3'], // '>=1.2.0 <1.3.0'
+      ['1', '1.0.0', '1.2.3', '1 <=1.2.3'], // '>=1.0.0 <2.0.0'
+
+      ['^1', '1.0.0', '1.3.0', '^1 <=1.3.0'], // '>=1.0.0 <2.0.0'
+      ['^1.2', '1.2.0', '1.3.0', '^1.2 <=1.3.0'], // '>=1.2.0 <2.0.0'
+      ['^1.2.3', '1.2.3', '1.3.0', '^1.2.3 <=1.3.0'], // '>=1.2.3 <2.0.0'
+
+      ['~>1', '1.0.0', '1.2.0', '~>1 <=1.2.0'], // '>=1.0.0 <2.0.0'
+      ['~>1.2', '1.2.0', '1.3.0', '>=1.2.0 <=1.3.0'], // '>=1.2.0 <1.3.0'
+      ['~>1.2.3', '1.2.3', '1.3.0', '>=1.2.3 <=1.3.0'], // '>=1.2.3 <1.3.0'
+
+      ['~1', '1.0.0', '1.2.0', '~1 <=1.2.0'], // '>=1.0.0 <2.0.0'
+      ['~1.2', '1.2.0', '1.3.0', '>=1.2.0 <=1.3.0'], // '>=1.2.0 <1.3.0'
+      ['~1.2.3', '1.2.3', '1.3.0', '>=1.2.3 <=1.3.0'], // '>=1.2.3 <1.3.0'
+
+      ['1.*', '1.0.0', '1.2.3', '1.* <=1.2.3'], // '>=1.0.0 <2.0.0'
+      ['1.*', '1.0.0', '2.3.4', '>=1.0.0 <=2.3.4'], // '>=1.0.0 <2.0.0'
+      ['1.2.*', '1.2.0', '1.2.3', '1.2.* <=1.2.3'], // '>=1.0.0 <1.3.0'
+      ['1.2.*', '1.0.0', '2.3.4', '>=1.2.0 <=2.3.4'], // '>=1.0.0 <1.3.0'
+
+      ['1.x', '1.0.0', '1.2.3', '1.x <=1.2.3'], // '>=1.0.0 <2.0.0'
+      ['1.x', '1.0.0', '2.3.4', '>=1.0.0 <=2.3.4'], // '>=1.0.0 <2.0.0'
+      ['1.2.x', '1.2.0', '1.2.3', '1.2.x <=1.2.3'], // '>=1.0.0 <1.3.0'
+      ['1.2.x', '1.0.0', '2.3.4', '>=1.2.0 <=2.3.4'], // '>=1.0.0 <1.3.0'
+
+      ['>=1.0.0 <=1.2.3', '1.0.0', '1.2.4', '>=1.0.0 <=1.2.4'],
+      [
+        '>=1.0.0 <=1.2.3 || >=1.3.0',
+        '1.0.0',
+        '1.4.0',
+        '>=1.0.0 <=1.2.3 || >=1.3.0 <=1.4.0',
+      ],
+      [
+        '>=1.0.0 <=1.2.3 || 1.3.0',
+        '1.0.0',
+        '1.4.0',
+        '>=1.0.0 <=1.2.3 || 1.4.0',
+      ],
+
+      ['>=1.0.0 <=1.2.3', '1.0.0', '1.5.0', '>=1.0.0 <=1.5.0'],
+      ['^1.0', '1.0.0', '1.0.7', '^1.0 <=1.0.7'],
+      ['^1', '1.0.0', '1.0.7-prerelease.1', '^1 <=1.0.7-prerelease.1'],
+      ['^1.0.0', '1.0.0', '1.0.7', '^1.0.0 <=1.0.7'],
+      ['~>1.0.0', '1.0.0', '1.1.7', '>=1.0.0 <=1.1.7'],
+      ['^1.0', '1.0.0', '1.1.7', '^1.0 <=1.1.7'],
+      ['~1.0', '1.0.0', '1.1.7', '>=1.0.0 <=1.1.7'],
+      ['~1.0', '1.0.0', '1.0.7-prerelease.1', '~1.0 <=1.0.7-prerelease.1'],
+      ['^1', '1.0.0', '2.1.7', '>=1.0.0 <=2.1.7'],
+      ['~1', '1.0.0', '1.1.7', '~1 <=1.1.7'],
+      ['5', '5.0.0', '5.1.7', '5 <=5.1.7'],
+      ['5.0', '5.0.0', '5.0.7', '5.0 <=5.0.7'],
+      ['5.0', '5.0.0', '5.1.7', '>=5.0.0 <=5.1.7'],
+      ['>=1.0.0', '1.0.0', '1.1.0', '>=1.0.0 <=1.1.0'],
+      ['>=1.0.0', '1.0.0', '1.1.0', '>=1.0.0 <=1.1.0'],
+      ['1.0.*', '1.0.0', '1.1.0', '>=1.0.0 <=1.1.0'],
+      ['1.*', '1.0.0', '2.1.0', '>=1.0.0 <=2.1.0'],
+      ['~0.6.1', '0.6.8', '0.7.0-rc.2', '~0.6.1 <=0.7.0-rc.2'],
+      ['5.0 || 5.5', '5.0.0', '6.1.7', '5.0 || >=5.5.0 <=6.1.7'],
```

Comment:
This is nearly impossible to guess right, and unlikely to happen in the wild. I almost feel like skipping it. Technically it would need to be something like: `(5.0 <=5.0.2) || (5.5 <= 5.5.4) || (6.1 <=6.1.7)` and I don't even know if that's valid.